templates/cli/lib/config.js.twig

```diff
@@ -408,59 +412,140 @@ class Global extends Config {
+
+    addLogin(login, data) {
+        this.set(login, data);
+    }
+
+    removeLogin(login, data) {
+        this.delete(login);
+    }
+
+    getEmail() {
+        if (!this.hasFrom(Global.PREFERENCE_EMAIL)) {
+            return "";
+        }
+
+        return this.getFrom(Global.PREFERENCE_EMAIL);
+    }
+
+    setEmail(email) {
+        this.setTo(Global.PREFERENCE_EMAIL, email);
+    }
+
     getEndpoint() {
-        if (!this.has(Global.PREFERENCE_ENDPOINT)) {
+        if (!this.hasFrom(Global.PREFERENCE_ENDPOINT)) {
             return "";
         }
-        return this.get(Global.PREFERENCE_ENDPOINT);
+
+        return this.getFrom(Global.PREFERENCE_ENDPOINT);
     }
 
     setEndpoint(endpoint) {
-        this.set(Global.PREFERENCE_ENDPOINT, endpoint);
+        this.setTo(Global.PREFERENCE_ENDPOINT, endpoint);
     }
 
     getSelfSigned() {
-        if (!this.has(Global.PREFERENCE_SELF_SIGNED)) {
+        if (!this.hasFrom(Global.PREFERENCE_SELF_SIGNED)) {
             return false;
         }
-        return this.get(Global.PREFERENCE_SELF_SIGNED);
+        return this.getFrom(Global.PREFERENCE_SELF_SIGNED);
     }
 
     setSelfSigned(selfSigned) {
-        this.set(Global.PREFERENCE_SELF_SIGNED, selfSigned);
+        this.setTo(Global.PREFERENCE_SELF_SIGNED, selfSigned);
     }
 
     getCookie() {
-        if (!this.has(Global.PREFERENCE_COOKIE)) {
+        if (!this.hasFrom(Global.PREFERENCE_COOKIE)) {
             return "";
         }
-        return this.get(Global.PREFERENCE_COOKIE);
+        return this.getFrom(Global.PREFERENCE_COOKIE);
     }
 
     setCookie(cookie) {
-        this.set(Global.PREFERENCE_COOKIE, cookie);
+        this.setTo(Global.PREFERENCE_COOKIE, cookie);
     }
 
     getProject() {
-        if (!this.has(Global.PREFERENCE_PROJECT)) {
+        if (!this.hasFrom(Global.PREFERENCE_PROJECT)) {
             return "";
         }
-        return this.get(Global.PREFERENCE_PROJECT);
+        return this.getFrom(Global.PREFERENCE_PROJECT);
     }
 
     setProject(project) {
-        this.set(Global.PREFERENCE_PROJECT, project);
+        this.setTo(Global.PREFERENCE_PROJECT, project);
     }
 
     getKey() {
-        if (!this.has(Global.PREFERENCE_KEY)) {
+        if (!this.hasFrom(Global.PREFERENCE_KEY)) {
             return "";
         }
-        return this.get(Global.PREFERENCE_KEY);
+        return this.getFrom(Global.PREFERENCE_KEY);
     }
 
     setKey(key) {
-        this.set(Global.PREFERENCE_KEY, key);
+        this.setTo(Global.PREFERENCE_KEY, key);
+    }
+
+
+    hasFrom(key) {
+        const current = this.getCurrentLogin();
+
+        if (current) {
+            const config = this.get(current);
+
+            return config[key] !== undefined;
+        }
+    }
+
+    getFrom(key) {
+        const current = this.getCurrentLogin();
+
+        if (current) {
+            const config = this.get(current);
+
+            return config[key];
+        }
+    }
+
+    setTo(key, value) {
```

Comment:
what exactly is the need for these new methods? how is it different from the `has` method that already exists?

Reply:
This is requires as each one of the multiple account is inside a an object, while the `has` method  will fetch the first level the `hasFrom` will fetch the first-level for a given account, 

### old json
```json
{
  "endpoint": "http://localhost/v1",
  "cookie": "cookie"
}
```


### new json
```json
{
  "6659ece200af3158fff95": {
    "endpoint": "http://cloud.appwrite.io/v1",
    "cookie": "Cookie",
    "email": "user@email.com"
  },
  "6659ece20016e58abe6b": {
    "endpoint": "http://localhost/v1",
    "cookie": "Cookie",
    "email": "legacy"
  },
  "current": "6659ece20016e58abe6b"
}
```